File: src/main/java/com/rs/java/game/area/banks/YanilleBank.java
package com.rs.java.game.area.banks;

import com.rs.java.game.WorldTile;
import com.rs.java.game.area.Area;
import com.rs.java.game.area.Shape;
import com.rs.java.game.area.shapes.Polygon;

public final class YanilleBank extends Area {

	@Override
	public Area update() {
		return this;
	}

	@Override
	public String name() {
		return "Yanille Bank";
	}

	@Override
	public Shape[] shapes() {
		return new Shape[] { new Polygon(new WorldTile[] {
				new WorldTile(2609, 3088, 0),
				new WorldTile(2609, 3097, 0),
				new WorldTile(2616, 3097, 0),
				new WorldTile(2616, 3088, 0),
				new WorldTile(2609, 3088, 0),
		}) };
	}

	@Override
	public boolean member() {
		return false;
	}

	@Override
	public Environment environment() {
		return Environment.SAFEZONE;
	}

}

File: src/main/java/com/rs/kotlin/game/world/area/areas/banks/YanilleBank.kt
package com.rs.kotlin.game.world.area.areas.banks

import com.rs.java.game.WorldTile
import com.rs.kotlin.game.world.area.Area
import com.rs.kotlin.game.world.area.shape.Polygon

class YanilleBank : Area(
    Polygon(
        arrayOf(
            WorldTile(2609, 3088, 0),
            WorldTile(2609, 3097, 0),
            WorldTile(2616, 3097, 0),
            WorldTile(2616, 3088, 0),
            WorldTile(2609, 3088, 0),
        )
    )
) {
    override fun update(): Area {
        return this
    }

    override fun name(): String {
        return "Yanille Bank"
    }

    override fun member(): Boolean {
        return false
    }

    override fun environment(): Environment {
        return Environment.SAFEZONE
    }
}
